### PR TITLE
avm2: URLStream support for connected and HTTP_STATUS event

### PR DESCRIPTION
**Pull request by evilpie**

### CODE REVIEW

File: core/src/avm2/globals/flash/net/URLStream.as [hunk 1]
```diff
@@ -74,10 +80,12 @@ package flash.net {
 
         public function load(request:URLRequest):void {
             this._loader.load(request);
+            this._connected = true;
```
Comment (adrian17) [inline core/src/avm2/globals/flash/net/URLStream.as:85]: I'm surprised it's this easy. Does it work this way even if the request fails (for example, the URLRequest url 404s)?

Comment (evilpie) [reply core/src/avm2/globals/flash/net/URLStream.as:85]: Yes. `connected` seems to stay true no matter if the request succeeds or fails.

**Review by adrian17 — APPROVED** [review]
> Yes. connected seems to stay true no matter if the request succeeds or fails.

Approved, but this is not really true - if the request throws immediately (say, a local file request on an swf without appropriate SandboxType), it stays `false`. I'm not sure if we already support this example or not.

**Comment by evilpie** [comment]
> > Yes. connected seems to stay true no matter if the request succeeds or fails.
> 
> Approved, but this is not really true - if the request throws immediately (say, a local file request on an swf without appropriate SandboxType), it stays `false`. I'm not sure if we already support this example or not.

If `this._loader.load(request)` throws `connected` would stay `false`. However it seems like our implementation of the URLLoader will never throw synchronously, because we always spawn `load_data_into_url_loader`.

**Pull request merged**